### DISCUSSION TITLE
how to assign an IP address to a switch

### DISCUSSION THREAD

**Discussion by ld06000** [netbox-community/netbox#13026] [Help Wanted!]
### NetBox version

v3.5.4

### Feature type

Data model extension

### Proposed functionality

So far I cannot find any possibility to assign an IP@ to a device expect by an interface which can be configured as primary 

### Use case

But for a switch which is not a level 3 device but a level 2 one, most of the time you cann access to the management through any interface. So thatwould be very convenient to be allowed to assign an IP@ to it...

**Comment by candlerb** [comment]
You need to create a virtual interface to which the IP address is attached. Many vendors already use a named interface for this, e.g. "Vlan1" on a Cisco. Otherwise, just make up your own name for it.

**Comment by ld06000** [comment]
Thanks a lot !!!
